systemtest/src/test/java/io/strimzi/systemtest/CruiseControlST.java

```diff
@@ -4,28 +4,254 @@
+    }
+
+    @Test
+    void testConfigurationReflection() throws IOException {
+        Pod cruiseControlPod = kubeClient().listPodsByPrefixInName(CRUISE_CONTROL_POD_PREFIX).get(0);
+
+        String cruiseControlPodName = cruiseControlPod.getMetadata().getName();
+
+        String configurationFileContent = cmdKubeClient().execInPod(cruiseControlPodName, "/bin/bash", "-c", "cat " + CRUISE_CONTROL_CONFIGURATION_FILE_PATH).out();
 
-    public static final String NAMESPACE = "cruise-control-test";
+        InputStream configurationFileStream = new ByteArrayInputStream(configurationFileContent.getBytes(StandardCharsets.UTF_8));
+
+        Properties fileConfiguration = new Properties();
+        fileConfiguration.load(configurationFileStream);
+
+        Container cruiseControlContainer = null;
+
+        for (Container container : cruiseControlPod.getSpec().getContainers()) {
+            if (container.getName().equals("cruise-control")) {
+                cruiseControlContainer = container;
+            }
+        }
+
+        EnvVar cruiseControlConfiguration = null;
+
+        for (EnvVar envVar : Objects.requireNonNull(cruiseControlContainer).getEnv()) {
+            if (envVar.getName().equals("CRUISE_CONTROL_CONFIGURATION")) {
+                cruiseControlConfiguration = envVar;
+            }
+        }
+
+        InputStream configurationContainerStream = new ByteArrayInputStream(Objects.requireNonNull(cruiseControlConfiguration).getValue().getBytes(StandardCharsets.UTF_8));
+
+        Properties containerConfiguration = new Properties();
+        containerConfiguration.load(configurationContainerStream);
+
+        LOGGER.info("Verifying that all configuration in the cruise control container matching the cruise control file {} properties", CRUISE_CONTROL_CONFIGURATION_FILE_PATH);
+
+        assertThat(containerConfiguration.getProperty("num.partition.metrics.windows"), is(fileConfiguration.getProperty("num.partition.metrics.windows")));
+        assertThat(containerConfiguration.getProperty("completed.user.task.retention.time.ms"), is(fileConfiguration.getProperty("completed.user.task.retention.time.ms")));
+        assertThat(containerConfiguration.getProperty("num.broker.metrics.windows"), is(fileConfiguration.getProperty("num.broker.metrics.windows")));
+        assertThat(containerConfiguration.getProperty("broker.metrics.window.ms"), is(fileConfiguration.getProperty("broker.metrics.window.ms")));
+        assertThat(containerConfiguration.getProperty("default.goals"), is(fileConfiguration.getProperty("default.goals")));
+        assertThat(containerConfiguration.getProperty("partition.metrics.window.ms"), is(fileConfiguration.getProperty("partition.metrics.window.ms")));
+        assertThat(containerConfiguration.getProperty("goals"), is(fileConfiguration.getProperty("goals")));
+
+        // TODO: is there some way to fetch the configuration in the Kafka CR?? (it is default) -> but where i can find it
```

Comment:
As above, you could `grep` the CC log.

Reply:
is this only way to grep the log @tomncooper . I do not like this solution is there any alternative? If no, I will use this log but in general logs in the containers are not so reliable.

Reply:
There are no default settings for CC specified in the Kafka CR, but you could add/override CC settings in the Kafka CR and check that they match the generated CC properties file in the container